django/core/mail/__init__.py

```diff
@@ -40,16 +40,30 @@
 ]
 
 
-def get_connection(backend=None, fail_silently=False, **kwds):
+def get_connection(backend=None, provider=None, fail_silently=False, **kwds):
     """Load an email backend and return an instance of it.
 
-    If backend is None (default), use settings.EMAIL_BACKEND.
+    If backend is None (default), use settings.EMAIL_PROVIDERS[provider]["BACKEND"].
+    If provider is None as well, use settings.EMAIL_PROVIDERS["default"]["BACKEND"].
 
     Both fail_silently and other keyword arguments are used in the
     constructor of the backend.
     """
-    klass = import_string(backend or settings.EMAIL_BACKEND)
-    return klass(fail_silently=fail_silently, **kwds)
+    if backend:
+        if provider:
+            raise ValueError(
+                "backend and provider are mutually exclusive, "
+                "so only use either of those arguments."
+            )
+        klass = import_string(backend)
+        return klass(fail_silently=fail_silently, **kwds)
+    provider = provider or "default"
+    klass = import_string(settings.EMAIL_PROVIDERS[provider]["BACKEND"])
+    return klass(
+        fail_silently=fail_silently,
+        use_localtime=settings.EMAIL_PROVIDERS[provider]["USE_LOCALTIME"],
```

Comment:
Because in my opinion this should not be part of the connection settings. Instead it should rather be part of the email generation process. We might further discuss this.